source/features/preserve-text-messages.js

```diff
@@ -0,0 +1,165 @@
+import debounce from 'lodash.debounce';
+import pick from 'lodash.pick';
+
+import {
+	observeEl,
+	getFromLocalStorage,
+	setToLocalStorage
+} from '../libs/utils';
+
+let DMDialogOpen = false;
+
+const DMModalSelector = '#dm_dialog';
+const DMConversationSelector = '.DMConversation';
+const DMTextBoxSelector = '#tweet-box-dm-conversation';
+
+export async function onDMDialogOpen() {
```

Comment:
@filipekiss would doing that make sense? Because they are main functions handling the logic for preserving messages. Perhaps they could be named better. Naming is hard 😅

Reply:
@gdad-s-river Yeah, the main point and what made me think about leaving them as they are for now is because they're too specific. The ideal would be to make them "wattchers" like `onSingleTweetOpen` and other functions like that, that receive a callback that runs when the conditions are triggered. Rename them to `preserveMessageDraft` and `deleteMessageDraft`, perhaps?

Reply:
> Rename them to `preserveMessageDraft` and `deleteMessageDraft`, perhaps?

Yes! I'll do that.

>  The ideal would be to make them "watchers"

Shall I do that as well?

Reply:
@gdad-s-river Sure! If you have any doubts, let me know and I can help you!